packages/components/src/toolbar-group/style.scss

```diff
@@ -21,11 +21,16 @@
 	min-height: $block-toolbar-height;
 	margin: 0;
 	border: $border-width solid $gray-900;
-	border-radius: $radius-block-ui;
 	background-color: $white;
 	display: inline-flex;
 	flex-shrink: 0;
 	flex-wrap: wrap;
+
+	// Unset for nested toolbars. Increase specificity.
+	& .components-toolbar.components-toolbar {
+		border-width: 0;
+		margin: 0;
+	}
```

Comment:
This snippet is already used in the new toolbar. This code also applies it to the legacy toolbar.